docs/pywbemcli/mock_support.rst

```diff
@@ -51,50 +51,103 @@ The following types of files are supported for the ``--mock-server`` option:
 
 .. index:: triple: Python files; server mock; add_cimobjects()
 
-* **Python files**: If the file extension is ``.py``, the file is considered
-  a Python file. The file is executed using Python's ``exec()`` (i.e. with
-  module namespace ``__builtin__``), and with the following Python global
-  variables made available:
-
-  * ``CONN`` (:class:`pywbem_mock.FakedWBEMConnection`):
-    This object provides a connection to the mock WBEM server. The methods
-    of this object can be used to create and modify CIM objects in the
-    mock repository.
-
-  * ``SERVER`` (:class:`pywbem.WBEMServer`):
-    This object is layered on top of the ``CONN`` object and provides access
-    to higher level features of the mock WBEM server, such as getting the
-    Interop namespace, adding namespaces, or building more complex objects
-    for the mock repository.
-
-  * ``VERBOSE`` (bool):
-    A flag that contains the value of the boolean
-    :ref:`--verbose general option` of pywbemcli.
-
-  The Python script can for example create Python objects of type
+* **Mock scripts**: If the file extension is ``.py``, the file is considered
+  a Python mock script.
+
+  Mock scripts can be used for any kind of setup of the mock WBEM server, for
+  example for creating namespaces, implementing and registering providers, or
+  adding CIM objects either from the corresponding Python objects or by
+  compiling MOF files or MOF strings.
+
+  Mock scripts support two approaches for passing the mock WBEM server they
+  should operate on:
+
+  * New-style: The mock script has a ``setup()`` function. This is the
+    recommended approach, but it is supported only on Python >=3.5. It enables
+    the mock environment of a connection definition to be cached.
+
+    New-style mock scripts are imported as a Python module into Python namespace
+    ``pywbemtools.pywbemcli.mockscripts.<mock-script-name>`` and their
+    ``setup()`` function is called. That function has the following interface:
+
+    .. code-block:: python
+
+        def setup(conn, server, verbose):
+            . . .
+
+    * ``conn`` (:class:`pywbem_mock.FakedWBEMConnection`):
+      This object provides a connection to the mock WBEM server. The methods
+      of this object can be used to create and modify CIM objects in the
+      mock repository, and for registering providers.
```

Comment:
change "and for registering" to "and to register"

Reply:
DONE